src/engine-actions/create-channels.js

```diff
@@ -0,0 +1,186 @@
+  const balanceCommon = Big(balance).div(quantumsPerCommon).toString()
+  const totalFeesEstimate = Big(feeEstimate).times(channelsToOpen)
+  const fundingAmountWithFees = amount.plus(totalFeesEstimate)
+
+  logger.debug('Received balance information', {
+    balance,
+    fundingAmountWithFees: fundingAmountWithFees.toString()
+  })
+
+  if (fundingAmountWithFees.gt(balance)) {
+    logger.error('Requested channel balance (plus fees) is larger than uncommitted balance', {
+      balance,
+      fundingAmountWithFees: fundingAmountWithFees.toString(),
+      fundingAmount,
+      symbol
+    })
+
+    // Find maximum estimated capacity given the fees we expect to pay
+    const maxFundingAmount = Big(balance).minus(totalFeesEstimate).div(quantumsPerCommon).toString()
+
+    throw new Error(`Requested funding amount exceeds available balance (${balanceCommon} ${symbol}). ` +
+      `Maximum estimated funding amount: ${maxFundingAmount} ${symbol}.`)
+  }
+}
+
+/**
+ * Executes a connection and opens channels w/ another lnd instance
+ *
+ * @param {string} paymentChannelNetworkAddress
+ * @param {string} fundingAmount - int64 string
+ * @param {number} roundBehavior - Behavior when encountering a channel that is too small
+ * @returns {Promise<void>} resolves void on success
+ */
+async function createChannels (paymentChannelNetworkAddress, fundingAmount, roundBehavior = CHANNEL_ROUNDING.DOWN) {
+  const {
+    feeEstimate,
+    minChannelBalance,
+    maxChannelBalance,
+    symbol,
+    logger,
+    client
+  } = this
+
+  if (!Object.values(CHANNEL_ROUNDING).includes(roundBehavior)) {
+    throw new Error(`Invalid round behavior: ${roundBehavior}. Must be one of ${Object.values(CHANNEL_ROUNDING)}.`)
+  }
+
+  if (!feeEstimate) {
+    throw new Error(`Currency configuration for ${symbol} has not been setup with a fee estimate`)
+  }
+
+  if (!maxChannelBalance) {
+    throw new Error(`Currency configuration for ${symbol} has not been setup with a max channel balance`)
+  }
+
+  if (!minChannelBalance) {
+    throw new Error(`Currency configuration for ${symbol} has not been setup with a min channel balance`)
+  }
+
+  // modify our funding amount based on the number of channels we're opening and the desired rounding behavior
+  let amount = getAmountForChannels(this, fundingAmount, roundBehavior)
+
+  // Check that we have sufficient balance to cover all of our channels
+  await assertBalance(this, amount)
+
+  // round up to find the number of channels to open
+  const channelsToOpen = amount.div(maxChannelBalance).round(0, 3)
+
+  const { publicKey, host } = networkAddressFormatter.parse(paymentChannelNetworkAddress)
+  const loggablePublicKey = loggablePubKey(publicKey)
+
+  logger.debug(`Attempting to create ${channelsToOpen} channels with ${loggablePublicKey}`)
+
+  await connectPeer(publicKey, host, { client, logger })
+
+  logger.debug(`Successfully connected to peer: ${loggablePublicKey}`)
+
+  for (var i = 0; i < channelsToOpen; i++) {
```

Comment:
I overheard part of a conversation that we use a `for` loop here instead of a `Promise.all` since the behavior could get weird trying to open multiple channels concurrently.

If that is the reasoning for using the `for` loop, it might be worth adding a comment.